src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -868,6 +886,11 @@ private static double ComputeWeightedDistance(ForStatementSyntax left, ForStatem
             {
                 distance = distance * 0.4 + localsDistance * 0.6;
             }
+            if (TryComputeLocalsDistance(left.Deconstruction, right.Deconstruction, out localsDistance))
```

Comment:
This is now getting tricky. One can make an edit of a for statement that changes the declarations to deconstructions and we might want to account for that (left node has Declaration and right node has Deconstruction, and vice versa).

For example,
```C#
for (var a = F().A, b = F().B; ...)  {}
```
changed to
```C#
for (var (a, b) = F(); ...)  { }
```

If the user makes such change while stepping thru the body of the loop we probably want to map the "new" variables to the ones that were there previously, since they are of the same names and in the same scope. If we don't do that the variables loose their values after the edit.

In order to match the variables we need to match the for statements. ```TryComputeLocalsDistance``` correlates the given nodes based on all variables declared in those nodes. It should do so regardless of whether the variables were defined by deconstruction or not.